dpnp/dpnp_algo/dpnp_algo.pyx

```diff
@@ -125,10 +126,16 @@ cpdef dparray dpnp_array(obj, dtype=None):
 
 
 cpdef dparray dpnp_astype(dparray array1, dtype_target):
-    cdef dparray result = dparray(array1.shape, dtype=dtype_target)
+    cdef DPNPFuncType param1_type = dpnp_dtype_to_DPNPFuncType(array1.dtype)
+    cdef DPNPFuncType param2_type = dpnp_dtype_to_DPNPFuncType(dtype_target)
```

Comment:
`dtype_target` might be diffrent. It can contain some object with no `dtype`. I think this is the cause of the error you saw